src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RefreshOrchestratorTest.java

```diff
@@ -267,16 +275,31 @@ void refreshRoleAssignmentRecords_nullUserRequest_feignException() {
     void nullJobIdTest_validate() {
         UserRequest userRequest = TestDataBuilder.buildUserRequest();
         String errorMessage = "Invalid JobId request";
+        Mockito.when(securityUtils.getServiceName())
+                .thenReturn("am_role_assignment_refresh_batch");
 
         BadRequestException exception = assertThrows(BadRequestException.class, () ->
                 sut.validate(null, userRequest));
         assertTrue(exception.getLocalizedMessage().contains(errorMessage));
+    }
+
+    @Test
+    @DisplayName("invalidServiceTokenTest_validate")
+    void invalidServiceTokenTest_validate() {
+        String errorMessage = "Invoking service is not permitted to call the Refresh API";
+        Mockito.doNothing().when(parseRequestService).validateUserRequest(any());
+        Mockito.when(securityUtils.getServiceName()).thenReturn("ccd_gw");
 
+        BadRequestException exception = assertThrows(BadRequestException.class, () ->
```

Comment:
See sonar report.  you may need to move `TestDataBuilder.buildUserRequest()` into a variable outside of assert function.